SNLS_testdriver_powell.cc

```diff
@@ -0,0 +1,350 @@
+#include <cstdlib>
```

Comment:
This seems excessively duplicative. If SNLS_NN_INDX and SNLSTRDLDG_J_INDX become the same, then it seems like you could use some compile-time settings to run the Broyden or ChebyQuad test problem either for the SNLSTrDlDenseG solver or for the SNLSHybrdDenseG solver. That is, if the idea is that this is a solver library and users can swap among solvers relatively readily, then you could set a good example of how that would look.

How about setting up a test problem per source file and then having compile-time settings for which solver gets used?

Reply:
I think that should be easy enough to do. It will just require a little bit of refactoring of the test problems. I'll include this as part of some other refactoring.